### PR TITLE
NMS-13421: BMP monitoring documentation diagram update

### PR DESCRIPTION
**Pull request by Bonrob2**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13421
* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13532

### CODE REVIEW

File: docs/modules/reference/pages/telemetryd/protocols/bmp.adoc [hunk 1]
```diff
@@ -5,7 +5,7 @@ The BGP Monitoring Protocol (BMP) provides a convenient interface for obtaining
 The integration in {page-component-title} lets you use these route views, status updates, and statistics for monitoring and management.
 
 .BMP integration overview
-image:bmp/bmp_integration_overview.png[BMP integration]
+image:bmp/bgp-monitoring.png[BMP integration]
```
Comment (j-white) [inline docs/modules/reference/pages/telemetryd/protocols/bmp.adoc:8]: The diagram needs some updates: We no longer go through Kafka when writing to PostgreSQL and the 'BMP-Postgres' component has been superseded by the `BMP-Persisting-Adapter` adapter.

Comment (Bonrob2) [reply docs/modules/reference/pages/telemetryd/protocols/bmp.adoc:8]: Thanks, @j-white . The whole chapter will need to be revised, as there are a lot of things that have changed. Ronny is currently working on the updates.

**Review by indigo423 — COMMENTED** [review]
The changes in the snmp-index.adoc seem to me not related to the BMP docs update?

**Comment by mmahacek** [comment]
> The changes in the snmp-index.adoc seem to me not related to the BMP docs update?

Those are actually review notes from another PR that got merged.

File: docs/modules/operation/pages/performance-data-collection/snmp-index.adoc [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 This section describes how to create a custom resource type to collect SNMP data from tables with arbitrary indexes:
 
-* <<resourcetype-def, create a resourceType definition>>
-* <<group-def, create "group" and "mibObj" definitions>>
-* <<system-def, add the group to a system definition>>
-* <<report-snmp-graph, create a report in snmp-graph.properties>>
+* <<resourcetype-def, Create a resourceType definition>>
+* <<group-def, Create "group" and "mibObj" definitions>>
+* <<system-def, Add the group to a system definition>>
+* <<report-snmp-graph, Create a report in snmp-graph.properties>>
```
Comment (mmahacek) [inline docs/modules/operation/pages/performance-data-collection/snmp-index.adoc:11]:
```suggestion
```

File: docs/modules/operation/pages/performance-data-collection/snmp-index.adoc [hunk 3]
```diff
@@ -1,13 +1,13 @@
 
 [[snmp-index]]
-= Custom resource types
+= Create a Custom Resource Type
 
 This section describes how to create a custom resource type to collect SNMP data from tables with arbitrary indexes:
```
Comment (mmahacek) [inline docs/modules/operation/pages/performance-data-collection/snmp-index.adoc:6]: The docs already builds a page-level table of contents on the right. Don't need to have a separate list of page sections here.
```suggestion
This section describes how to create a custom resource type to collect SNMP data from tables with arbitrary indexes.
```

**Comment by Bonrob2** [comment]
@j-white, @mmahacek, or @indigo423 looking for approval to merge.

**Comment by j-white** [comment]
Minor nit on the .png: the "BMP Persisting Adapter" could use a "Telemetryd adapter" icon, rather than having a specific "Persisting Adapter" icon

**Comment by Bonrob2** [comment]
> Minor nit on the .png: the "BMP Persisting Adapter" could use a "Telemetryd adapter" icon, rather than having a specific "Persisting Adapter" icon

Updated!

**Comment by mmahacek** [comment]
> Minor nit on the .png: the "BMP Persisting Adapter" could use a "Telemetryd adapter" icon, rather than having a specific "Persisting Adapter" icon

On that point, the icons for listener/parser/adapter are attached to text boxes that have those same words, making the icons/legend redundant.  And the parsing adapter is using Title Case while all others are UPPER CASE.  Is there a reason for that?

**Comment by Bonrob2** [comment]
> > Minor nit on the .png: the "BMP Persisting Adapter" could use a "Telemetryd adapter" icon, rather than having a specific "Persisting Adapter" icon
> 
> On that point, the icons for listener/parser/adapter are attached to text boxes that have those same words, making the icons/legend redundant. And the parsing adapter is using Title Case while all others are UPPER CASE. Is there a reason for that?

Parsing adapter should be UC. My oversight. The icons specifically indicate "telemetryd adapter", while the boxes show different implementations of it, so I think the legend still makes sense. Will update the case for parsing adapter.

**Review by mmahacek — APPROVED** [review]
The graphic looks good to me, though I can't verify its accuracy.

**Comment by indigo423** [comment]
@Bonrob2 I've updated the diagram, we can probably reuse it for flows and other telemetry things as well.

**Review by Bonrob2 — COMMENTED** [review]
The graphic looks great. Only question is if we need to do something with the protobuf to indicate it's part of OpenNMS?

**Review by indigo423 — APPROVED** [review]
BMP diagram updated and there is a dedicated commit which fixes the image:: syntax.

**Review by Bonrob2 — COMMENTED** [review]
LGTM!

**Pull request merged**